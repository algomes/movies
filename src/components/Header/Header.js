import './Header.scss';

import React from 'react';
import {
    Link as RouterLink
} from "react-router-dom";

const Header = () => {
    return (
        <div className="navbar">
          <RouterLink className="link" to="/">
            <span className="text" variant="h5">
              Home
            </span>
          </RouterLink>
          <RouterLink className="link" to="/movies">
            <span className="text" variant="h5">
              Movies
            </span>
          </RouterLink>
        </div>
    );
  }
  
  export default Header;