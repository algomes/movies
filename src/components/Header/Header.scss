.navbar {
    display: flex;
    justify-content: flex-end;
    background-color: black;
    top: 0;
}

.link {
    margin: 1rem;
    text-decoration: none;
}

.text {
    color: white;
    font-family: Lucida Handwriting,cursive;
}