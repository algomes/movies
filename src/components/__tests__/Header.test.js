import Header from '../Header/Header';
import {
  BrowserRouter as Router,
} from "react-router-dom";
import { mount } from 'enzyme';

//Testando com snapshot pois testei o Router no App.js
test('snapshot from Header component', () => {
  const wrapper = mount(
    <Router>
      <Header/>
    </Router>
  );

  expect(wrapper.debug()).toMatchSnapshot();
});