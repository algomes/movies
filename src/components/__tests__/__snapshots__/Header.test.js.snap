// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot from Header component 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <Header>
      <div className=\\"navbar\\">
        <Link className=\\"link\\" to=\\"/\\">
          <LinkAnchor className=\\"link\\" href=\\"/\\" navigate={[Function: navigate]}>
            <a className=\\"link\\" href=\\"/\\" onClick={[Function: onClick]}>
              <span className=\\"text\\" variant=\\"h5\\">
                Home
              </span>
            </a>
          </LinkAnchor>
        </Link>
        <Link className=\\"link\\" to=\\"/movies\\">
          <LinkAnchor className=\\"link\\" href=\\"/movies\\" navigate={[Function: navigate]}>
            <a className=\\"link\\" href=\\"/movies\\" onClick={[Function: onClick]}>
              <span className=\\"text\\" variant=\\"h5\\">
                Movies
              </span>
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </Header>
  </Router>
</BrowserRouter>"
`;
